Calendar/View Controllers/CalendarViewController.m

```diff
@@ -72,7 +137,68 @@ - (void)didTapCancel {
 
 - (void)didTapCreateWithEvent:(nonnull Event *)event {
     [self dismissViewControllerAnimated:YES completion:nil];
-    [self.parseHandler uploadToParseWithEvent:event];
+    [self.parseHandler uploadToParseWithEvent:event completion:^(Event * _Nonnull event, NSDate * _Nonnull date, NSString * _Nullable error) {
+        if (error) {
+            [self failedRequestWithMessage:error];
+        } else {
+            [self successfullyUploadedEvent:event forDate:date];
+        }
+    }];
+}
+
+- (void)didTapClose {
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (void)didDeleteEvent:(Event *)event {
+    [self.dateLogicHandler deleteEvent:event];
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (CGFloat)collectionView:(UICollectionView *)collectionView layout:(UICollectionViewLayout *)collectionViewLayout minimumLineSpacingForSectionAtIndex:(NSInteger)section {
+    return 0;
+}
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
+    return [self.dateLogicHandler getNumberOfElements];
+}
+
+- (__kindof UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+    ScheduleCollectionCell *cell = [collectionView
+                                    dequeueReusableCellWithReuseIdentifier:@"scheduleCellId"
+                                    forIndexPath:indexPath];
+    NSDate *date = [self.dateLogicHandler getDateForIndexPath:indexPath];
+    [self.scheduleDecorator decorateBaseScheduleWithDate:date contentView:cell.scheduleView];
+    [cell.scheduleView.scrollView setContentOffset:CGPointMake(0, self.pageContentOffsetY) animated:false];
```

Comment:
This is clever. I think it needs to be reset at some point though when transitioning between some types of views.